DeleteMiddleNode.py

```diff
@@ -0,0 +1,36 @@
+class ListNode:
+    def __init__(self, value=0, next=None):
+        self.value = value
+        self.next = next
+
+def deleteNode(node):
+    if node is None or node.next is None:
```

Comment:
Be careful using `is` over `==` in Python. `is` is used for [identity equality](https://www.blog.pythonlibrary.org/2017/02/28/python-101-equality-vs-identity/#:~:text=Hopefully%20by%20now%20you%20understand,object%20you%20are%20referring%20to.). Also, since `None` is already falsey in Python, you don't need to explicitly check against it.
```suggestion
    if not node or not node.next:
```

Reply:
Also, is there a reason you don't delete the node if there's only one node in the list?